src/pages/alerts.astro

```diff
@@ -6,7 +6,7 @@ const alertsRoute = getAPIRoute("ws");
 
 import "../styles/alerts.scss";
 
-const token = Astro.cookies.get("token").value ?? "";
+const token = Astro.cookies.get("token");
 
 if (!token) {
   return Astro.redirect("/login");
```

Comment:
No, because the [page].astro redirect will only work if you're navigating to a page that doesn't exist. It's basically a replacement for the 404 page. It doesn't work if you're navigating to an existing page. You can test that in the preview by navigating to /profile without being logged in, since the redirect is commented out for now.